homeassistant/components/slide_local/coordinator.py

```diff
@@ -0,0 +1,127 @@
+"""DataUpdateCoordinator for slide_local integration."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import TYPE_CHECKING, Any
+
+from goslideapi.goslideapi import (
+    AuthenticationFailed,
+    ClientConnectionError,
+    ClientTimeoutError,
+    DigestAuthCalcError,
+    GoSlideLocal as SlideLocalApi,
+)
+
+from homeassistant.const import (
+    CONF_API_VERSION,
+    CONF_HOST,
+    CONF_MAC,
+    CONF_PASSWORD,
+    STATE_CLOSED,
+    STATE_CLOSING,
+    STATE_OPEN,
+    STATE_OPENING,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DEFAULT_OFFSET, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+if TYPE_CHECKING:
+    from . import SlideConfigEntry
+
+
+class SlideCoordinator(DataUpdateCoordinator[dict[str, Any]]):
+    """Get and update the latest data."""
+
+    def __init__(self, hass: HomeAssistant, entry: SlideConfigEntry) -> None:
+        """Initialize the data object."""
+        super().__init__(
+            hass, _LOGGER, name="Slide", update_interval=timedelta(seconds=15)
+        )
+        self.slide = SlideLocalApi()
+        self.api_version = entry.data[CONF_API_VERSION]
+        self.mac = entry.data[CONF_MAC]
+        self.host = entry.data[CONF_HOST]
+        self.password = entry.data[CONF_PASSWORD]
+
+    async def _async_setup(self) -> None:
+        """Do initialization logic for Slide coordinator."""
+        _LOGGER.debug("Initializing Slide coordinator")
+
+        await self.slide.slide_add(
+            self.host,
+            self.password,
+            self.api_version,
+        )
+
+        try:
+            data = await self.slide.slide_info(self.host)
+        except (ClientConnectionError, ClientTimeoutError) as ex:
+            raise ConfigEntryNotReady(
+                translation_domain=DOMAIN,
+                translation_key="config_entry_not_ready",
+            ) from ex
+
+        if data is None or (data.get(CONF_MAC) is None and self.mac == ""):
```

Comment:
When is data None?

Reply:
Similar to the mac-question, I got this from the CC: https://github.com/ualex73/slide/blob/214c44be4f60ac071e1c44694bfd5b83ffeaa16d/custom_components/slide/cover.py#L383